crates/red_knot_python_semantic/resources/mdtest/snapshots/membership_test.md_-_Comparison___Membership_Test_-_Return_type_that_doesn't_implement_`__bool__`_correctly.snap

```diff
@@ -28,7 +28,7 @@ mdtest path: crates/red_knot_python_semantic/resources/mdtest/comparison/instanc
 # Diagnostics
 
 ```
-error: lint:unsupported-bool-conversion: Boolean conversion is unsupported for type `NotBoolable`; its `__bool__` method isn't callable
+error: lint:unsupported-bool-conversion: Boolean conversion is unsupported for type `NotBoolable`
```

Comment:
This can remain a TODO, but really this diagnostic should explain why we are looking at `NotBoolable` in the first place (because we did an `in` operation on `WithContains`, and its `__contains__` returns `NotBoolable`

Reply:
Ah yeah it looks like this is even mentioned in the corresponding mdtest:

``````
TODO: Ideally the message would explain to the user what's wrong. E.g,

```ignore
error: [operator] cannot use `in` operator on object of type `WithContains`
    note: This is because the `in` operator implicitly calls `WithContains.__contains__`, but `WithContains.__contains__` is invalidly defined
    note: `WithContains.__contains__` is invalidly defined because it returns an instance of `NotBoolable`, which cannot be evaluated in a boolean context
    note: `NotBoolable` cannot be evaluated in a boolean context because its `__bool__` attribute is not callable
```
``````

That would be a really nice diagnostic!

Reply:
I think I'd prefer to fix that as part of tackling https://github.com/astral-sh/ruff/issues/16371#issuecomment-2682014149. This error should be propagated up so that it's reported as an `[operator]` error rather than a `[not-boolable]` error; that will allow us to provide more details in the diagnostic, but to do that we need to cleanup some of our internals a little bit.